src/parser.ts

```diff
@@ -111,9 +110,12 @@ export class Parser extends DiagnosticEmitter {
   dependees: Map<string, Source> = new Map();
 
   /** Constructs a new parser. */
-  constructor(program: Program) {
-    super(program.diagnostics);
-    this.program = program;
+  constructor(
+    public diagnostics: DiagnosticMessage[] = [],
```

Comment:
This re-redefines the base class's field, so should remove the `public` modifier.

Reply:
Just a clarification question:

I should remove the `public` keyword and set the `this.diagnostics` property. right?

Reply:
Just removing the `public` keywords should be enough, since the argument is forward to the `super` constructor which sets `this.diagnostics`.

Reply:
All set!